src/main/java/com/google/netpcapanalysis/dao/FrequencyDaoImpl.java

```diff
@@ -0,0 +1,127 @@
+package com.google.netpcapanalysis.dao;
+
+import com.google.netpcapanalysis.models.Flagged;
+import com.google.netpcapanalysis.models.PCAPdata;
+import java.util.*; 
+import java.io.*;
+import java.lang.*;
+
+import io.pkts.PacketHandler;
+import io.pkts.Pcap;
+import io.pkts.buffer.Buffer;
+import io.pkts.packet.Packet;
+import io.pkts.packet.TCPPacket;
+import io.pkts.packet.UDPPacket;
+import io.pkts.packet.IPPacket;
+import io.pkts.protocol.Protocol;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import com.google.netpcapanalysis.interfaces.dao.PCAPDao;
+import com.google.netpcapanalysis.interfaces.dao.PCAPParserDao;
+import com.google.netpcapanalysis.interfaces.dao.FrequencyDao;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class FrequencyDaoImpl implements FrequencyDao {
+  private ArrayList<PCAPdata> allPCAP = new ArrayList<PCAPdata>(); 
+  private ArrayList<PCAPdata> finalFreq = new ArrayList<PCAPdata>();
+  private HashMap<String, PCAPdata> finalMap = new HashMap<String, PCAPdata>();
+  private String myip = "";
+  private String filename;  
+  private boolean first = true;
+
+  public FrequencyDaoImpl(ArrayList<PCAPdata> packets) {
+    allPCAP = packets; 
+  }
+
+  /* Retrieve all necessary entities, processes, and puts into final arraylist */
+  public void loadFrequency() {
+    findMyIP();
+    processData();
+    putFinalFreq();
+  }
+
+  public ArrayList<PCAPdata> getAllPCAP() {
+    return allPCAP;
+  }
+
+  /* Find local ip address based on highest recurring IP address */
+  private void findMyIP() {
```

Comment:
Does your PCAP data contain the external address for the users IP or is it internal? If it's internal you could maybe just scan for which IP's are in the common range for home routers like:
10.0.0.0        -   10.255.255.255
172.16.0.0      -   172.31.255.255
192.168.0.0     -   192.168.255.255